wolfcrypt/src/aes.c

```diff
@@ -4464,12 +4464,12 @@ static void roll_auth(Aes* aes, const byte* in, word32 inSz, byte* out)
     word32 remainder;
 
     /* encode the length in */
-    if (inSz <= 0xFEFF) {
+    if (inSz <= 0xFEFF) { /* 16-bit */
         authLenSz = 2;
         out[0] ^= ((inSz & 0xFF00) >> 8);
         out[1] ^=  (inSz & 0x00FF);
     }
-    else if (inSz <= 0xFFFFFFFF) {
+    else if (inSz <= 0xFFFFFF) { /* 24-bit */
```

Comment:
My bad, mis-read that code and will revert. The compiler warned that this check is always true, which it is and this could just be else here for now.